tests/snapshots/test_sampling_with_sample_rate_1_and_rate_limit_3_and_rule_0.json

```diff
@@ -14,7 +14,6 @@
       "runtime-id": "9d1cfdf7026e45dea46fc4c83768cc7b"
     },
     "metrics": {
-      "_dd.limit_psr": 1,
```

Comment:
Here the sample rule overrides the default sample rate of 1 and the span is dropped by trace sample rules. We should NOT set `_dd.limit_psr` on spans that were not dropped by the rate limitter.